Swift/QtUI/QtXMLConsoleWidget.cpp

```diff
@@ -40,30 +45,34 @@ QtXMLConsoleWidget::QtXMLConsoleWidget() {
     buttonLayout->setContentsMargins(10,0,20,0);
     buttonLayout->setSpacing(0);
 
+    debugenabled = new QCheckBox(tr("Manually enter XML"), bottom);
     enabled = new QCheckBox(tr("Trace input/output"), bottom);
     enabled->setChecked(true);
     buttonLayout->addWidget(enabled);
 
     buttonLayout->addStretch();
 
-
-
     QPushButton* clearButton = new QPushButton(tr("Clear"), bottom);
-    QPushButton* sendXMLButton = new QPushButton(tr("Send XML"), bottom);
     connect(clearButton, SIGNAL(clicked()), textEdit, SLOT(clear()));
-    connect(sendXMLButton, SIGNAL(clicked()), this, SLOT(showWindow()));
+    connect(debugenabled, SIGNAL(toggled(bool)), this, SLOT(showWindow(bool)));
     buttonLayout->addWidget(clearButton);
-    buttonLayout->addWidget(sendXMLButton);
+    buttonLayout->addWidget(debugenabled);
     setWindowTitle(tr("Debug Console"));
     emit titleUpdated();
 }
 
 QtXMLConsoleWidget::~QtXMLConsoleWidget() {
 }
 
-void QtXMLConsoleWidget::showWindow() {
-    QtXMLSenderWidget* XMLWindow = new QtXMLSenderWidget();
-    XMLWindow->show();
+void QtXMLConsoleWidget::showWindow(bool showXMLSender) {
+    if(showXMLSender)
+    {
+        XMLWindow->show();
+    }
+    else
+    {
+        XMLWindow->hide();
+    }
```

Comment:
Instead of this if you could also pass a boolean flag to `QWidget::setVisible` method, see http://doc.qt.io/qt-5/qwidget.html#visible-prop .